File: api/src/endpoint/organization/add_member.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::OrganizationMemberResource};

pub struct AddMember;

impl Endpoint for AddMember {
    const PATH: &'static str = "/organization/{org_name}/repositories";
    const METHOD: http::Method = http::Method::POST;

    type Request = AddMemberRequest;
    type Response = AddMemberResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct AddMemberRequest {
    pub user_name: String,
    pub role: String,
}

pub type AddMemberResponse = OrganizationMemberResource;

File: gitdot-api-ts/src/endpoint/organization/add-member.ts
import { z } from "zod";

import { OrganizationMemberResource } from "../../resource";

export const AddMemberRequest = z.object({
  user_name: z.string(),
  role: z.string(),
});
export type AddMemberRequest = z.infer<typeof AddMemberRequest>;

export const AddMember = {
  path: "/organization/{org_name}/repositories",
  method: "POST",
  request: AddMemberRequest,
  response: OrganizationMemberResource,
} as const;
export type AddMember = typeof AddMember;
